### PR TITLE
feat(theme-classic): allow className as option for type: "search"

### PR DESCRIPTION
**Pull request by JPeer264**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
  - Could not find any tests for these changes. Please correct me if I am wrong.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Please refer to this issue comment
https://github.com/facebook/docusaurus/pull/7164#issuecomment-1118221792

## Test Plan

I tried to use `yarn lerna link` and linked it in a new setup project but still got `[ERROR] ValidationError: "navbar.items[0].className" is not allowed` although it has been added to the Joi validation schema. I might need help on this part (cc: @slorber)

### Test links

Deploy preview: https://deploy-preview-7357--docusaurus-2.netlify.app/
New changes: https://deploy-preview-7357--docusaurus-2.netlify.app/docs/api/themes/configuration#navbar-search

### CODE REVIEW

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Please add a test case for the validation schema, although it's kind of obvious that it will work. Unit tests are always more effective than `yarn link` 😉

**Comment by JPeer264** [comment]
@Josh-Cena I would be more than happy to add one, but could not find any tests there. Are there some existing which I overlooked? I am pretty sure I am not the first person adding tests for this

**Comment by Josh-Cena** [comment]
The tests are here: https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/__tests__/validateThemeConfig.test.ts

**Comment by JPeer264** [comment]
Oh ya true. Just found them thanks, I will add them right away.

**Comment by JPeer264** [comment]
@Josh-Cena implementation wise (apart from the validation) there are no tests am I right?

**Comment by Josh-Cena** [comment]
No, we don't have theme tests yet😞 I think this is the best we can do. It doesn't look like it shouldn't work.

File: packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx [hunk 1]
```diff
@@ -75,7 +76,7 @@ export default function NavbarContent(): JSX.Element {
           <NavbarItems items={rightItems} />
           <NavbarColorModeToggle className={styles.colorModeToggle} />
           {autoAddSearchBar && (
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx:78]: No need for an extra variable now. Just `searchBarItem && (...)`

Comment (JPeer264) [reply packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx]: Ok also my preferred way, wasn't quite sure if you want to preserve the naming there.

File: packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx [hunk 2]
```diff
@@ -74,8 +74,8 @@ export default function NavbarContent(): JSX.Element {
         <>
           <NavbarItems items={rightItems} />
           <NavbarColorModeToggle className={styles.colorModeToggle} />
-          {autoAddSearchBar && (
-            <NavbarSearch>
+          {searchBarItem && (
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx:77]: Ah, wait, my bad. The logic should be the other way: `!searchBarItem && (...)`. This is when the user doesn't specify a search item explicitly and we "auto add" it.

Comment (JPeer264) [reply packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx:77]: Oh totally true, I missed that part. Nice catch!

Comment (JPeer264) [reply packages/docusaurus-theme-classic/src/theme/Navbar/Content/index.tsx]: Then I got the logic wrong anyways. Updated in a fixup commit

**Comment by JPeer264** [comment]
@Josh-Cena friendly ping :)

**Review by Josh-Cena — APPROVED** [review]
Yes, it's good from my side. However I'd also like to get @slorber's approval. He'd been unavailable for the past two weeks; I think he'll be back this week.

File: packages/docusaurus-theme-classic/src/validateThemeConfig.ts [hunk 3]
```diff
@@ -202,6 +202,7 @@ const LocaleDropdownNavbarItemSchema = NavbarItemBaseSchema.append({
 
 const SearchItemSchema = Joi.object({
   type: Joi.string().equal('search').required(),
+  className: Joi.string().default(''),
```
Comment (slorber) [inline packages/docusaurus-theme-classic/src/validateThemeConfig.ts:205]:
```suggestion
  className: Joi.string(),
```

I'd remove the default value?

For consistency with the Html item also having a className + the TS declaration where className is optional

**Review by slorber — APPROVED** [review]
LGTM 👍

**Comment by Josh-Cena** [comment]
Thanks @JPeer264!

**Pull request merged**